src/dialog-holder.component.ts

```diff
@@ -29,84 +29,102 @@ export class DialogHolderComponent {
   constructor(private resolver: ComponentFactoryResolver) {}
 
   /**
-   * Adds dialog
+   * Configures then adds dialog to the dialogs array, and populates with data passed in
    * @param {Type<DialogComponent>} component
    * @param {object?} data
    * @param {DialogOptions?} options
    * @return {Observable<*>}
    */
-  addDialog<T, T1>(component:Type<DialogComponent<T, T1>>, data?:T, options?:DialogOptions):Observable<T1> {
+  addDialog<T, T1>(component: Type<DialogComponent<T, T1>>, data?: T, options?: DialogOptions): Observable<T1> {
     options = options || <DialogOptions>{};
-    let factory = this.resolver.resolveComponentFactory(DialogWrapperComponent);
-    let componentRef = this.element.createComponent(factory, options.index);
-    let dialogWrapper: DialogWrapperComponent = <DialogWrapperComponent> componentRef.instance;
-    let _component: DialogComponent<T,T1> =  dialogWrapper.addComponent(component);
-    if(typeof(options.index) !== 'undefined') {
+
+    const factory = this.resolver.resolveComponentFactory(DialogWrapperComponent);
+    const componentRef = this.viewContainer.createComponent(factory, options.index);
+    const dialogWrapper: DialogWrapperComponent = <DialogWrapperComponent> componentRef.instance;
+    const _component: DialogComponent<T, T1> =  dialogWrapper.addComponent(component);
+
+    if (typeof(options.index) !== 'undefined') {
       this.dialogs.splice(options.index, 0, _component);
-    }
-    else {
+    } else {
       this.dialogs.push(_component);
     }
-    setTimeout(()=>{
-      dialogWrapper.container.nativeElement.classList.add('show');
-      dialogWrapper.container.nativeElement.classList.add('in');
+    setTimeout(() => {
+      dialogWrapper.wrapper.nativeElement.classList.add('show');
+      dialogWrapper.wrapper.nativeElement.classList.add('in');
     });
-    if(options.autoCloseTimeout) {
-      setTimeout(()=>{
+    if (options.autoCloseTimeout) {
+      setTimeout(() => {
         this.removeDialog(_component);
       }, options.autoCloseTimeout);
     }
-    if(options.closeByClickingOutside) {
-      dialogWrapper.closeByClickOutside();
+    if (options.closeByClickingOutside) {
+      dialogWrapper.onClickOutsideModalContent( () => {
+        this.removeDialog(dialogWrapper.content);
+      });
     }
-    if(options.backdropColor) {
-      dialogWrapper.container.nativeElement.style.backgroundColor = options.backdropColor;
+    if (options.backdropColor) {
```

Comment:
Do we need this type of configuration? I think this could be set on the css files. Also, it's easir o maintin the layout if the project uses sass, for example, where the colors are set via variables.

Reply:
I agree - i don't use it either will remove this